deepchem/models/torch_model.py

```diff
@@ -0,0 +1,1124 @@
+    """
+    self._ensure_built()
+    dataset = NumpyDataset(X, y, w)
+    return self.fit(
+        dataset,
+        nb_epoch=1,
+        max_checkpoints_to_keep=max_checkpoints_to_keep,
+        checkpoint_interval=self._global_step + 2 if checkpoint else 0,
+        variables=variables,
+        loss=loss,
+        callbacks=callbacks)
+
+  def _predict(
+      self, generator: Iterable[Tuple[Any, Any, Any]],
+      transformers: List[Transformer], uncertainty: bool,
+      other_output_types: Optional[OneOrMany[str]]) -> OneOrMany[np.ndarray]:
+    """
+    Predict outputs for data provided by a generator.
+
+    This is the private implementation of prediction.  Do not
+    call it directly.  Instead call one of the public prediction
+    methods.
+
+    Parameters
+    ----------
+    generator: generator
+      this should generate batches, each represented as a tuple of the form
+      (inputs, labels, weights).
+    transformers: list of dc.trans.Transformers
+      Transformers that the input data has been transformed by.  The output
+      is passed through these transformers to undo the transformations.
+    uncertainty: bool
+      specifies whether this is being called as part of estimating uncertainty.
+      If True, it sets the training flag so that dropout will be enabled, and
+      returns the values of the uncertainty outputs.
+    other_output_types: list, optional
+      Provides a list of other output_types (strings) to predict from model.
+    Returns:
+      a NumPy array of the model produces a single output, or a list of arrays
+      if it produces multiple outputs
+    """
+    results: Optional[List[np.ndarray]] = None
+    variances: Optional[List[np.ndarray]] = None
+    if uncertainty and (other_output_types is not None):
+      raise ValueError(
+          'This model cannot compute uncertainties and other output types simultaneously. Please invoke one at a time.'
+      )
+    if uncertainty:
+      if self._variance_outputs is None or len(self._variance_outputs) == 0:
+        raise ValueError('This model cannot compute uncertainties')
+      if len(self._variance_outputs) != len(self._prediction_outputs):
+        raise ValueError(
+            'The number of variances must exactly match the number of outputs')
+    if other_output_types:
+      if self._other_outputs is None or len(self._other_outputs) == 0:
+        raise ValueError(
+            'This model cannot compute other outputs since no other output_types were specified.'
+        )
+    self._ensure_built()
+    self.model.eval()
+    for batch in generator:
+      inputs, labels, weights = batch
+      inputs, _, _ = self._prepare_batch((inputs, None, None))
+
+      # Invoke the model.
+      if len(inputs) == 1:
+        inputs = inputs[0]
+      output_values = self.model(inputs)
+      if isinstance(output_values, torch.Tensor):
+        output_values = [output_values]
+      output_values = [t.detach().numpy() for t in output_values]
```

Comment:
I think this code will not work with GPU.

If we use GPU, we call `cpu()` before calling `numpy()`.

```python
def tensor2numpy(value):
    if torch.cuda.is_available():
        return value.detach().cpu().numpy()
    else:
        return value.detach().numpy()

output_values = [tensor2numpy(t) for t in output_values]

```